### PR TITLE
PHPORM-239 Convert `_id` and `UTCDateTime` in results of `Model::raw()` before hydratation

### LINKED ISSUE #3151: _id returned by raw()
- Laravel-mongodb Version: 5.0.0
- PHP Version: 8.2.22
- Database Driver & Version: PECL mongodb driver 1.19.3 

### Description: 
Following the latest update and id management, raw requests do not seem to return 'id' but still return '_id'

### Steps to reproduce
1. $tmp = \App\Models\Container::first()->toArray(); => this returns id
2. $tmp = \App\Models\Container::raw(function($collection) {
        return $collection->findOne();
    }); => this returns _id

### Expected behaviour
Should receive 'id'

### Actual behaviour
Receive '_id'

### PR DESCRIPTION
**Pull request by GromNaN**
Fix PHPORM-239
Fix #3151

### Checklist

- [x] Add tests and ensure they pass
- [x] Add an entry to the CHANGELOG.md file

### CODE REVIEW

**Comment by GromNaN** [comment]
I don't see any way to change this behavior without breaking backward compatibility.

File: src/Eloquent/Builder.php [hunk 1]
```diff
@@ -177,22 +180,27 @@ public function raw($value = null)
...
             return $this->model->hydrate($results);
         }
 
-        // Convert MongoDB BSONDocument to a single object.
-        if ($results instanceof BSONDocument) {
-            $results = $results->getArrayCopy();
-
-            return $this->model->newFromBuilder((array) $results);
+        // Convert MongoDB Document to a single object.
+        if (is_object($results) && (property_exists($results, '_id') || property_exists($results, 'id'))) {
```
Comment (alcaeus) [inline src/Eloquent/Builder.php:191]: Confirmed that PHP correctly calls the `has_property` handler when using `property_exists`: https://github.com/php/php-src/blob/1be989bbf0e46db8438384db184fca424392cfef/Zend/zend_builtin_functions.c#L1020..L1023

File: tests/ModelTest.php [hunk 2]
```diff
@@ -924,6 +920,59 @@ public function testRaw(): void
         $this->assertNotNull($result);
     }
 
+    #[DataProvider('provideTypeMap')]
+    public function testRawHyradeModel(array $typeMap): void
```
Comment (alcaeus) [inline tests/ModelTest.php:924]: Typo:
```suggestion
    public function testRawHydrateModel(array $typeMap): void
```

File: tests/ModelTest.php [hunk 3]
```diff
@@ -924,6 +920,59 @@ public function testRaw(): void
...
+        $this->assertEquals([1, 2, 3], $user->list);
+    }
+
+    public static function provideTypeMap(): Generator
+    {
+        yield 'default' => [[]];
+        yield 'array' => [['root' => 'array', 'document' => 'array', 'array' => 'array']];
+        yield 'object' => [['root' => 'object', 'document' => 'object', 'array' => 'array']];
+        yield 'Library BSON' => [['root' => BSONDocument::class, 'document' => BSONDocument::class, 'array' => BSONArray::class]];
+        yield 'Driver BSON' => [['root' => 'bson', 'document' => 'bson', 'array' => 'bson']];
```
Comment (alcaeus) [inline tests/ModelTest.php:973]: Nit: when `root` is `bson`, the `document` and `array` types are ignored completely.
```suggestion
        yield 'Driver BSON' => [['root' => 'bson']];
```

**Comment by alcaeus** [comment]
LGTM, but tests need a little more work.

File: tests/ModelTest.php [hunk 4]
```diff
@@ -924,6 +920,59 @@ public function testRaw(): void
...
+            ],
+        ));
+
+        $this->assertInstanceOf(User::class, $user);
+        $this->assertArrayNotHasKey('_id', $user->getAttributes());
+        $this->assertArrayHasKey('id', $user->getAttributes());
+        $this->assertNotEmpty($user->id);
+        $this->assertInstanceOf(Carbon::class, $user->now);
+        $this->assertEquals(['foo' => 'bar'], (array) $user->embed);
+        $this->assertEquals([1, 2, 3], (array) $user->list);
```
Comment (GromNaN) [inline tests/ModelTest.php:947]: We don't convert nested documents and array into array for single result. This is easily done for cursor and BSON documents with typemap.

File: tests/Query/BuilderTest.php [hunk 5]
```diff
@@ -1381,6 +1381,31 @@ function (Builder $elemMatchQuery): void {
...
+                                ],
+                            ],
+                            ['age' => 15],
+                        ],
+                    ],
+                    [], // options
+                ],
+            ],
+            fn (Builder $builder) => $builder->where([
+                '$or' => [
```
Comment (GromNaN) [inline tests/Query/BuilderTest.php:1403]: Added this test case to ensure `DB::where($filter)->get()` is identical to `$collection->find($filter)` (for the doc team).

**Review by alcaeus — APPROVED** [review]

**Pull request merged**